pkg/server/datastore/datastore.go

```diff
@@ -35,6 +35,7 @@ type DataStore interface {
 	CreateOrReturnRegistrationEntry(context.Context, *common.RegistrationEntry) (*common.RegistrationEntry, bool, error)
 	DeleteRegistrationEntry(ctx context.Context, entryID string) (*common.RegistrationEntry, error)
 	FetchRegistrationEntry(ctx context.Context, entryID string) (*common.RegistrationEntry, error)
+	FetchRegistrationEntries(ctx context.Context, entryIDs []string) ([]*common.RegistrationEntry, error)
```

Comment:
I wonder if returning a map wouldn't be a better interface. I think you're likely to be interested in both the found entries and the entries that were not found (no longer in the database).

Reply:
Yea i think so.

Reply:
This change is in